src/Framework/FileMock.php

```diff
@@ -40,16 +48,51 @@ public static function create($content, $extension = NULL)
 	}
 
 
+	public static function register()
+	{
+		if (!self::$registered) {
+			stream_wrapper_register(self::PROTOCOL, __CLASS__);
+			self::$registered = TRUE;
+		}
+	}
+
+
 	public function stream_open($path, $mode)
 	{
+		if (!preg_match('#^([rwaxc]).*?(\+)?#', $mode, $m)) {
```

Comment:
Not sure, how PCRE should be strict. I tried, that first letter must match mode, others can be mess.